docs: consolidate models (#5200)
langchain-ai/langgraph: docs/docs/agents/models.md

# Models

> LangGraph provides built-in support for LLMs (language models) via the LangChain library. This makes it easy to integrate various LLMs into your agents and workflows.

Reviewer: ```suggestion
LangGraph provides built-in support for [LLMs (language models)](https://python.langchain.com/docs/concepts/chat_models/) via the LangChain library. This makes it easy to integrate various LLMs into your agents and workflows.
```

> ## Initializing a model

Reviewer: ```suggestion
## Initialize a model
```

Refer to the [API reference](https://python.langchain.com/api_reference/core/language_models/langchain_core.language_models.chat_models.BaseChatModel.html#langchain_core.language_models.chat_models.BaseChatModel.disable_streaming) for more information on `disable_streaming`

> ### Adding model fallbacks

Reviewer: ```suggestion
### Add model fallbacks
```

See this [guide](https://python.langchain.com/docs/how_to/fallbacks/#fallback-to-better-model) for more information on model fallbacks.

### Rate limiting

> Langchain includes a built-in in memory rate limiter. This rate limiter is thread safe and can be shared by multiple threads in the same process.

Reviewer: ```suggestion
Langchain includes a built-in in-memory rate limiter. This rate limiter is thread safe and can be shared by multiple threads in the same process.
```

```python
from langchain_core.rate_limiters import InMemoryRateLimiter
from langchain_anthropic import ChatAnthropic

rate_limiter = InMemoryRateLimiter(
    requests_per_second=0.1,  # <-- Super slow! We can only make a request once every 10 seconds!!
    check_every_n_seconds=0.1,  # Wake up every 100 ms to check whether allowed to make a request,
    max_bucket_size=10,  # Controls the maximum burst size.
)

model = ChatAnthropic(
   model_name="claude-3-opus-20240229", 
   rate_limiter=rate_limiter
)
```

> - [Handle rate limiting](https://python.langchain.com/docs/how_to/chat_model_rate_limiting/): Add rate limiting to your chat model to handle API rate limits gracefully.

Reviewer: ```suggestion
See the LangChain docs for more information on how to [handle rate limiting](https://python.langchain.com/docs/how_to/chat_model_rate_limiting/).
```
